src/chocolatey.resources/helpers/functions/Get-EnvironmentVariable.ps1

```diff
@@ -10,11 +10,27 @@
 # distributed under the License is distributed on an "AS IS" BASIS,
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
-# limitations under the License.
-
-function Get-EnvironmentVariable([string] $Name, [System.EnvironmentVariableTarget] $Scope) {
-    [Environment]::GetEnvironmentVariable($Name, $Scope)
-}
-
-# Some enhancements to think about here.
-# $machinePath = [Microsoft.Win32.Registry]::LocalMachine.OpenSubKey("SYSTEM\CurrentControlSet\Control\Session Manager\Environment\").GetValue("PATH", "", [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames).ToString();
+# limitations under the License.
+
+function Get-EnvironmentVariable([string] $Name, [System.EnvironmentVariableTarget] $Scope, [bool] $PreserveVariables = $False) {
+    if ($pathType -eq [System.EnvironmentVariableTarget]::Machine) {
+        $reg = [Microsoft.Win32.Registry]::Machine.OpenSubKey("Environment", $true)
+    } else {
+        $reg = [Microsoft.Win32.Registry]::CurrentUser.OpenSubKey("Environment", $true)
+    }
+
+    if ($PreserveVariables -eq $True) {
+        $option = [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames
+    } else {
+        $option = [Microsoft.Win32.RegistryValueOptions]::None
+    }
+
+    $value = $reg.GetValue('Path', $null, $option)
+
+    $reg.Close()

```

Comment:
RegistryKey objects wrap unmanaged resources (registry handles), so their usage should be guarded by try...finally:

````
<open the key>
try
{
    ...
}
finally
{
    $reg.Close()
}
````
Otherwise, the handles will be leaked in case of errors.